activation/wire/challenge_v2.go

```diff
@@ -0,0 +1,42 @@
+package wire
+
+import (
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/hash"
+	"go.uber.org/zap/zapcore"
+)
+
+//go:generate scalegen
+
+// NIPostChallengeV2 collects information needed to generate a PoET challenge.
+// It's hash is the challenge registered in the PoET.
+type NIPostChallengeV2 struct {
+	PublishEpoch     types.EpochID
+	PrevATXID        types.ATXID
+	PositioningATXID types.ATXID
+	CommitmentATXID  *types.ATXID
+	InitialPost      *PostV1
+}
+
+// Hash serializes the NIPostChallenge and returns its hash.
+// The serialized challenge is first prepended with a byte 0x00, and then hashed
+// for second preimage resistance of poet membership merkle tree.
+func (c *NIPostChallengeV2) Hash() types.Hash32 {
+	ncBytes := codec.MustEncode(c)
+	return hash.Sum([]byte{0x00}, ncBytes)
+}
```

Comment:
This should probably also be changed to be the root of a merkle tree formed by the `NiPostChallengeV2`, to ensure we can proof any field to be a specific value in a malfeasance proof without having to attach the full ATX. We just need to make sure that the `NiPostChallengeV2` is attached via its root to the tree to be able to extract each field individually.

Regarding pre-image resistance: we need to check if this is still needed with the new signature scheme.